airflow/providers/google/cloud/operators/dataflow.py

```diff
@@ -132,6 +132,15 @@ class DataflowCreateJavaJobOperator(BaseOperator):
     :type check_if_running: CheckJobRunning(IgnoreJob = do not check if running, FinishIfRunning=
         if job is running finish with nothing, WaitForRun= wait until job finished and the run job)
         ``jar``, ``options``, and ``job_name`` are templated so you can use variables in them.
+    :param wait_until_finished: (Optional)
```

Comment:
You made the description a bit complicated.

```
If True, wait for the end of pipeline execution before exiting. If False, it only waits for it to starts (``JOB_STATE_RUNNING``).

The default behavior  depends on the type of pipeline:
* for the streaming pipeline, wait for jobs to start,
* for the batch pipeline, wait for the jobs to complete.

.. warning::

    You cannot call ``PipelineResult.wait_until_finish`` method in your pipeline code for the operator to work properly. i. e. you must use asynchronous execution. Otherwise, your pipeline will always wait until finished. For more information, look at: `Asynchronous execution <https://cloud.google.com/dataflow/docs/guides/specifying-exec-params#python_10>`__
```